resteasy-servlet-initializer/src/main/java/org/jboss/resteasy/plugins/servlet/ResteasyServletInitializer.java

```diff
@@ -99,69 +91,82 @@ protected void handleNoApplicationClass(Set<Class<?>> providers, Set<Class<?>> r
    protected void register(Class<?> applicationClass, Set<Class<?>> providers, Set<Class<?>> resources, ServletContext servletContext)
    {
       ApplicationPath path = applicationClass.getAnnotation(ApplicationPath.class);
+      ServletRegistration.Dynamic reg;
+      String mapping;
+      String prefix;
       if (path == null)
       {
-         // todo we don't support this yet, i'm not sure if partial deployments are supported in all servlet containers
-         return;
+          if (servletContext.getServletRegistration(applicationClass.getName()) == null)
+          {
+              // Application subclass has no @ApplicationPath and no declared mappings to use
+              return;
+          }
+          reg = servletContext.addServlet(applicationClass.getName(), HttpServlet30Dispatcher.class);
+          Collection<String> mappings = reg.getMappings();
+          if (mappings == null || mappings.isEmpty())
+          {
+              // no declared mappings
+              return;
+          }
+          mapping = mappings.iterator().next();

```

Comment:
Related to my question on RESTEASY-2650, about using the same Application on two different paths, wouldn't you want to add a servlet for each path? Maybe I'm on the wrong track here.